04_setup_ironic.sh

```diff
@@ -73,14 +74,23 @@ for IMAGE_VAR in $(env | grep "_LOCAL_IMAGE=" | grep -o "^[^=]*") ; do
                 git fetch origin pull/${IMAGE_PR}/head:pr${IMAGE_PR}
                 git checkout pr${IMAGE_PR}
         fi
+        
+        # If we set extra packages, we add them to the main pkgs base file
+        EXTRA_PKGS_FILE_NAME=${IMAGE_VAR/_LOCAL_IMAGE}_EXTRA_PACKAGES
+        EXTRA_PKGS_FILE=${!EXTRA_PKGS_FILE_NAME:-}
+        if [[ -n $EXTRA_PKGS_FILE ]]; then
+            cp $EXTRA_PKGS_FILE "$REPOPATH"
```

Comment:
it was definitely not doing it when I started working on this 3 months ago :)
Thanks for checking, I added a basename command afterwards that should fix that

Reply:
Thanks, this fixes the second problem, but relative path names still aren't working

```
[derekh@u07 dev-scripts]$ IRONIC_EXTRA_PACKAGES=3eer ./04_setup_ironic.sh 
+(./04_setup_ironic.sh:78): main(): cp 3eer /home/derekh/ironic-image
cp: cannot stat '3eer': No such file or directory

```

Reply:
Relative path names cannot work as it is since we're changing dir into the image repo, unless the file is already present in the image repo, but then it would be useless to copy it there.
The idea is to have the file in a place that is not the image repo and then copy it inside the image repo.
I can add 'absolute' before 'path' to the comment and to the README to be more explicit.

Reply:
> Relative path names cannot work as it is since we're changing dir into the image repo, unless the file is already present in the image repo, but then it would be useless to copy it there.
> The idea is to have the file in a place that is not the image repo and then copy it inside the image repo.
> I can add 'absolute' before 'path' to the comment and to the README to be more explicit, or define in advance a path where the file should be located, removing some degree of freedom.

We could use OLDPWD and realpath to convert to an absolute path name, e.g.
`EXTRA_PKGS_FILE=$(cd $OLDPWD ; realpath $EXTRA_PKGS_FILE)`

but if that wont work for some reason I'm happy with mentioning it in config_example.sh

Reply:
doesn't this imply that the file would necessary be in $OLDPWD ?

Reply:
It implies that the EXTRA_PKGS_FILE is either absolute or is relative to  $OLDPWD, The script was run from $OLDPWD so either should work.

Reply:
so if EXTRA_PKGS_FILE is not in the dir where we run the script from (I suppose dev-scripts repo dir), it won't work
if we go for this solution we need to modify the comments and the README, which I don't mind doing
having the EXTRA_PKGS_FILE under dev-scripts was my first intention anyway

Reply:
> so if EXTRA_PKGS_FILE is not in the dir where we run the script from (I suppose dev-scripts repo dir), it won't work
> if we go for this solution we need to modify the comments and the README, which I don't mind doing
> having the EXTRA_PKGS_FILE under dev-scripts was my first intention anyway

I don't see the case where it wouldn't work, see examples below, once the path is correct relative to where the script was run its ok

```
[derekh@u07 dev-scripts]$ cat test.sh
EXTRA_PKGS_FILE=$1
cd /tmp
EXTRA_PKGS_FILE=$(cd $OLDPWD ; realpath $EXTRA_PKGS_FILE)
echo $EXTRA_PKGS_FILE

[derekh@u07 dev-scripts]$ bash test.sh afile
/home/derekh/dev-scripts/afile
[derekh@u07 dev-scripts]$ bash test.sh ../afile
/home/derekh/afile
[derekh@u07 dev-scripts]$ bash test.sh /tmp/afile
/tmp/afile
[derekh@u07 dev-scripts]$ bash test.sh ~/ironic-image/extra-package
/home/derekh/ironic-image/extra-package
```

Reply:
alright, I'll apply the change

Reply:
thanks